### DISCUSSION TITLE
Question about BERT vocab

### DISCUSSION THREAD

**Discussion by hyesulim** [seominjoon/kaist-ai605#48] [Q&A] [answered]
Hello, I came up with a question while using BERT vocab!!!
Although this question is not directly related to this course work, I thought this Discussions could answer my question :) 

According to [JAPANESE AND KOREAN VOICE SEARCH](https://static.googleusercontent.com/media/research.google.com/ko//pubs/archive/37842.pdf), WordPeice is created in the way of increasing the likelihood.
> Generate a new word unit by combining two units out of the current word inventory to increment the word unit inventory by one. Choose the new word unit out of all possible ones **_that increases the likelihood on the training data_** the most when added to the model.

However, I'm not sure if the tokens in published BERT vocab ([vocab.txt](https://huggingface.co/bert-base-uncased/blob/main/vocab.txt)) are sorted in the way of they are created (from maximum likelihood to minimum likelihood).

In summary, I'd like to know the ordering of tokens in the BERT vocab.

It would be so grateful if anyone can answer this question!!!
Thanks in advance.

**Comment by ggoggam** [comment, marked as answer]
I believe the tokens in `vocab.txt` are just mappings generated by WordPiece tokenizer, which is a type of byte-level Byte Pair Encoding algorithm. The vocab will not be sorted in the order of its likelihood, but rather arbitrary appearance from training data. You can read more about this [here](https://huggingface.co/transformers/tokenizer_summary.html#:~:text=WordPiece%20is%20the%20subword%20tokenization,is%20very%20similar%20to%20BPE.)
